src/sentry/migrations/0728_add_subscription_fk_to_incident.py

```diff
@@ -0,0 +1,87 @@
+# Generated by Django 5.0.6 on 2024-06-12 16:47
+
+import logging
+
+import django.db.models.deletion
+from django.db import migrations
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+
+import sentry.db.models.fields.foreignkey
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.utils.query import RangeQuerySetWrapper
+
+logger = logging.getLogger(__name__)
+
+
+def _backfill_incident_subscription(
+    apps: StateApps, schema_editor: BaseDatabaseSchemaEditor
+) -> None:
+    Incident = apps.get_model("sentry", "Incident")
+
+    # use RangeQuerySetWrapper to avoid loading all Incidents into memory
+    for incident in RangeQuerySetWrapper(Incident.objects.all()):
+        alert_rule = incident.alert_rule
+        if not alert_rule:
+            logger.info(
+                "Incident found with no alert_rule. skipping",
+                extra={"incident_id": incident.id},
+            )
+            continue
+
+        snuba_query = alert_rule.snuba_query
+        subscriptions = list(snuba_query.subscriptions.all())
+        if len(subscriptions) == 0 or len(subscriptions) > 1:
+            logger.info(
+                "Incident found with no or multiple subscriptions. skipping",
+                extra={
+                    "incident_id": incident.id,
+                    "snuba_query_id": snuba_query.id,
+                    "subscription_ids": [
+                        subscription.id for subscription in subscriptions if len(subscriptions) > 0
+                    ],
+                },
+            )
+            continue
+
+        subscription = subscriptions[0]
+        incident.subscription = subscription
+        incident.save()
+
+
+class Migration(CheckedMigration):
+    # This flag is used to mark that a migration shouldn't be automatically run in production.
+    # This should only be used for operations where it's safe to run the migration after your
+    # code has deployed. So this should not be used for most operations that alter the schema
+    # of a table.
+    # Here are some things that make sense to mark as post deployment:
+    # - Large data migrations. Typically we want these to be run manually so that they can be
+    #   monitored and not block the deploy for a long period of time while they run.
+    # - Adding indexes to large tables. Since this can take a long time, we'd generally prefer to
+    #   run this outside deployments so that we don't block them. Note that while adding an index
+    #   is a schema change, it's completely safe to run the operation after the code has deployed.
+    # Once deployed, run these manually via: https://develop.sentry.dev/database-migrations/#migration-deployment
+
+    is_post_deployment = False
```

Comment:
We have ~27M rows in the affected tables. The backfill should be a post-deploy migration, which means you'd need to separate it from the schema change.

Reply:
Aha that makes sense.
thanks for the catch!

I separated out the column add to a [separate PR here](https://github.com/getsentry/sentry/pull/72666)